urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/presentation/SearchListMovieAndCreateBooking_CLI.java

```diff
@@ -0,0 +1,474 @@
+		
+		for(Cineplex cineplex: cineplexBL.getCineplexes()) {
+			List<Cinema> cinemas = cineplex.getCinemas(movie);
+			
+			if(cinemas == null) {
+				continue;
+			}
+			
+			System.out.println(cineplex.getCineplexName());
+			System.out.println("..................................");
+			
+			// Go through every cinemas
+			for(Cinema cinema: cinemas) {
+				List<ShowTime> showTimes = cinema.getShowTimes(movie);
+				
+				if(showTimes == null) {
+					continue;
+				}
+					
+				Hashtable<String, List<ShowTime>> showTimeHashTable = new Hashtable<String, List<ShowTime>>();
+				//Preprocessing for showTime to follow the cathay format
+				SimpleDateFormat sdf = new SimpleDateFormat("d MMM, E");
+				for(ShowTime st: showTimes) {
+					String d = sdf.format(st.getTime());
+												
+					if(showTimeHashTable.get(d) == null) {
+						showTimeHashTable.put(d, new ArrayList<ShowTime>());
+					}
+					
+					showTimeHashTable.get(d).add(st);
+				}
+									
+				SimpleDateFormat timeformat = new SimpleDateFormat("HH:mm");
+				
+				for(Enumeration<String> e = showTimeHashTable.keys();e.hasMoreElements();) {
+					String dateString = e.nextElement();
+				
+					System.out.print(dateString);
+					
+					for(ShowTime showTime: showTimeHashTable.get(dateString)) {
+						// showTime can be accessed using the index. It is in the sequence it is printed.
+						// index starts at zero even though the index printed out start at 1. -1 before accessing.
+						_showTimes.add(showTime); 
+												
+						String time = new SimpleDateFormat("HH:mm").format(showTime.getTime());
+						System.out.print("\t[" + (index+1) + "] " + time);
+						
+						index++;
+					}
+					
+					System.out.println();
+				}
+				System.out.println();
+			}
+		}
+	}
+		
+	public void printSeats(ShowTime showTime) {
+		Cinema cinema = showTime.getCinema();
+		
+		System.out.println();
+		System.out.println("            SCREEN                  ");
+		System.out.println(" -------------------------------  ");
+		System.out.print(" ");
+		for(int i=1;i<=cinema.getNoOfSeatColumn();i++) {
+			System.out.print("  " + i);
+			
+			if(i == cinema.getNoOfSeatColumn()/2) {
+				System.out.print("     ");
+			}
+		}
+		System.out.println();
+		System.out.println(" -------------------------------  ");
+		String prev = "";
+		int seat_column_count = 0;
+		for(Seat seat: cinema.getSeats()) { 
+			if(!prev.equals(seat.getSeatRow())) {
+				seat_column_count = 0;
+				
+				if(!prev.equals("")){
+					System.out.println(" " + prev);
+					System.out.println(" -------------------------------  ");
+				}
+				System.out.print(seat.getSeatRow() + " ");
+			}
+			
+			if(seat.isOccupiedAt(showTime)) {
+				if(seat.getSeatType() == SeatType.COUPLE && seat.getSeatNo()%2 == 1) {
+					System.out.print("[X ");
+				}
+				else if(seat.getSeatType() == SeatType.COUPLE && seat.getSeatNo()%2 == 0) {
+					System.out.print(" X]");
+				}
+				else {
+					System.out.print("[X]");
+				}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 411:419, 422:432.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.